calculator/lexer.py

```diff
@@ -11,8 +45,88 @@ def lex(s):
       * "*" - Multiplication
       * "+" - Addition
       * "-" - Subtraction
+      * "/" - Division
     * The following grouping symbols are supported:
       * "("
       * ")"
     """
-    return []
+
+    # Defining char_before and char_after
+    char_before = None
+    char_after = None
+    # Partial_num is the var that stores my nums until I have to append it to my tokens list
+    partial_num = None
+    # List that stores the tokens
+    tokens = []
+    # Stripping the input line of spaces
+    stripped_line = line.replace(" ", "")
```

Comment:
This works for infix because numbers are always split by operators, but in other notation schemes (i.e. RPN) this is a bug.  For instance, if you want to lex the RPN string `1 1 +` you will end up with `11+` which will parse to `[11, "+"]`.  This same lexer should be able to be used for either.  

I would recommend that, rather than stripping the spaces, treat a space as an indication that your partial_num is completely seen.  Make sure you add a test case with:
* Multiple spaces: `lex("1 +   1 ") == [1, "+", 1]`
* RPN: `lex("1 1 +") == [1, 1, "+"]`

Reply:
👍